composeApp/src/androidMain/kotlin/org/ooni/probe/MainActivity.kt

```diff
@@ -1,18 +1,48 @@
 package org.ooni.probe
 
+import android.content.Intent
+import android.net.Uri
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
+import androidx.navigation.NavHostController
+import co.touchlab.kermit.Logger
 
 class MainActivity : ComponentActivity() {
+    private lateinit var navController: NavHostController
     private val app get() = applicationContext as AndroidApplication
 
     override fun onCreate(savedInstanceState: Bundle?) {
         enableEdgeToEdge()
         super.onCreate(savedInstanceState)
         setContent {
-            App(app.dependencies)
+            navController = App(app.dependencies)
```

Comment:
This would be ideal. `MainActivity#onNewIntent` is called when activity is in the backstack and `MainActivity#onCreate` is called when the app is not in the backstack

Reply:
Done